modules/reindex/src/main/java/org/elasticsearch/reindex/Reindexer.java

```diff
@@ -285,7 +300,7 @@ protected RequestWrapper<IndexRequest> buildRequest(ScrollableHitSource.Hit doc)
             }
 
             // id and source always come from the found doc. Scripts can change them but they operate on the index request.
-            index.id(doc.getId());
+            index.id(destinationIndexIdMapper.reindexId(doc.getId()));
```

Comment:
this will probably also speed reindex up

Reply:
I don't *think* so. It's a noop for standard indices. For tsdb indices it saves us some bits on the wire and a comparison. We still have to regenerate the id.